hooks/tk-hiero-export/hiero_translate_template.py

```diff
@@ -59,6 +59,19 @@ def execute(self, template, output_type, **kwargs):
         if output_type == 'script':
             template_str = template_str.replace('{output}', 'scene')
 
+        # Nuke Studio project string has version number which is an issue when we have to resolve template by path
+        # so replacing {name} with 'plate' string
+        # and stripping {output} to simplify template to {Sequence}_{Shot}_{Step}_{name}.v{tk_version}.mov
+        # engine specific template would have been useful here (as could update only for nuke studio)
+        if output_type == "plate":
+            template_str = template_str.replace('{name}', 'plate')
+            template_str = template_str.replace('{output}', '')
```

Comment:
I know the pre-existing code was using `String.replace()`, but this hook should really be using `Template.apply_fields()` instead since its significantly safer. The code would be something like this:

```
fields = self.parent.context.as_template_fields(template)

# Substitute the variables that Hiero will inject during export
hiero_fields = {
    "Sequence": "{sequence}",
    "Shot": "{shot}",
    "name": "{project}",
    "output": "{clip}",
    "width": "{width}",
    "height": "{height}",
    "version": "{tk_version}",
    "extension": "{ext}"
}
fields.update(hiero_fields)

for name, key in template.keys.iteritems():
    if isinstance(key, sgtk.templatekey.SequenceKey):
        fields.update({name: "FORMAT:#"})

if output_type == "script":
    fields.update({"output": "scene"})
elif output_type == "plate":
    fields.update({"name": "plate"})
    del fields["output"]

template_str = template.apply_fields(fields, ignore_types=hiero_fields.keys())
```

Reply:
Just realized `ignore_types` isn't exposed. You can use `_apply_fields()` instead for the time being, and I'll expose that param in a future release.